external/skiboot/edk2-compat-process.c

```diff
@@ -214,23 +216,34 @@ int get_auth_descriptor2(const void *buf, const size_t buflen, void **auth_buffe
 
 static bool validate_cert(char *signing_cert, int signing_cert_size)
 {
-	mbedtls_x509_crt x509;
+	//NICK CHILD removed direct mbedtls call, use general crypto
+	// mbedtls_x509_crt x509;
+	void *x509;
 	char *x509_buf = NULL;
 	int rc;
 
-	mbedtls_x509_crt_init(&x509);
-	rc = mbedtls_x509_crt_parse(&x509, (unsigned char *)signing_cert, signing_cert_size);
-
+	// NICK CHILD removed direct mbedtls call, use general crypto
+	// mbedtls_x509_crt_init(&x509);
+	// rc = mbedtls_x509_crt_parse(&x509, (unsigned char *)signing_cert, signing_cert_size);
 	/* If failure in parsing the certificate, exit */
-	if(rc) {
-		prlog(PR_ERR, "X509 certificate parsing failed %04x\n", rc);
+	// if(rc) {
+	// 	prlog(PR_ERR, "X509 certificate parsing failed %04x\n", rc);
+	x509 = crypto_x509_parse_der((unsigned char *)signing_cert, signing_cert_size);
+	if (!x509) {
+		prlog(PR_ERR, "X509 certificate parsing failed\n");
+
 		return false;
 	}
 
 	x509_buf = zalloc(CERT_BUFFER_SIZE);
-	rc = mbedtls_x509_crt_info(x509_buf, CERT_BUFFER_SIZE, "\tCRT:", &x509);
-
-	mbedtls_x509_crt_free(&x509);
+	// NICK CHILD removed direct mbedtls call, use general crypto
+	// rc = mbedtls_x509_crt_info(x509_buf, CERT_BUFFER_SIZE, "\tCRT:", &x509);
+	rc = crypto_x509_get_long_desc(x509_buf, CERT_BUFFER_SIZE, "CRT:", x509);
+	// NICK CHILD General Question: Should we be printing to debug here?
+	// it seems wierd to parse to info string then not print
+	// NICK CHILD removed direct mbedtls call, use general crypto
```

Comment:
mmm good point. not sure, will check in future when I extract this into a more generic library.